ui/components/app/snaps/snap-install-warning/snap-install-warning.js

```diff
@@ -50,47 +33,120 @@ export default function SnapInstallWarning({
   snapName,
 }) {
   const t = useI18nContext();
-  const [checkboxState, dispatch] = useReducer(checkboxStateReducer, {});
-
-  const isAllChecked = warnings.every((warning) => checkboxState[warning.id]);
-
-  const onCheckboxClicked = useCallback((checkboxId) => {
-    dispatch({ type: 'check', checkboxId });
-  }, []);
+  const [userAgree, setUserAgree] = useState(false);
 
   const SnapInstallWarningFooter = () => {
     return (
-      <div className="snap-install-warning__footer">
+      <Box
+        display={Display.Flex}
+        justifyContent={JustifyContent.center}
+        alignItems={AlignItems.center}
+        width={BlockSize.Full}
+      >
         <Button
-          className="snap-install-warning__footer-button"
           type="primary"
-          disabled={!isAllChecked}
+          disabled={!userAgree}
           onClick={onSubmit}
+          width={BlockSize.Full}
+          size={ButtonSize.Lg}
         >
           {t('confirm')}
         </Button>
-      </div>
+      </Box>
     );
   };
 
+  function constructWarningElementComponentArray(permissionWarnings) {
+    if (permissionWarnings.length === 1) {
+      return [
+        <Text
+          fontWeight={FontWeight.Medium}
+          as="span"
+          key="warningMessageSubject"
+        >
+          {permissionWarnings[0].warningMessageSubject}
+        </Text>,
+      ];
+    }
+
+    if (permissionWarnings.length === 2) {
+      const firstWarningSubject = permissionWarnings[0].warningMessageSubject;
+      const secondWarningSubject = permissionWarnings[1].warningMessageSubject;
+      return [
+        <Text
+          fontWeight={FontWeight.Medium}
+          as="span"
+          key="warningMessageSubject"
+        >
+          {t('andForTwoItems', [firstWarningSubject, secondWarningSubject])}
+        </Text>,
+      ];
+    }
+
+    return permissionWarnings.map((warning, index) => {
+      if (permissionWarnings.length - 1 === index) {
+        return [];
+      }
+      const elements = [];
```

Comment:
Could we just return an array in each of the if statements instead of pushing to one?